tests/two_d_array_int/two_d_array_int.c

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <inttypes.h>
+
+#include "macro_utils/macro_utils.h"
+#include "testrunnerswitcher.h"
+
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+
+#include "c_pal/interlocked.h"
+#include "c_pal/thandle.h"
+#include "c_pal/threadapi.h"
+
+#include "c_util/two_d_array.h"
+
+typedef struct TEST_THANDLE_TAG
+{
+    int64_t a;
+} TEST_THANDLE;
+
+TWO_D_ARRAY_DEFINE_STRUCT_TYPE(TEST_THANDLE);
+THANDLE_TYPE_DECLARE(TWO_D_ARRAY_TYPEDEF_NAME(TEST_THANDLE));
+THANDLE_TYPE_DEFINE(TWO_D_ARRAY_TYPEDEF_NAME(TEST_THANDLE));
+
+TWO_D_ARRAY_TYPE_DECLARE(TEST_THANDLE);
+TWO_D_ARRAY_TYPE_DEFINE(TEST_THANDLE);
```

Comment:
>TWO_D_ARRAY_TYPE_DEFINE [](http://example.com/codeflow?start=0&length=23)

would be nice if this implicitly did `TWO_D_ARRAY_DEFINE_STRUCT_TYPE` so the code that uses 2D array just has `TWO_D_ARRAY_TYPE_DECLARE` in the header and `TWO_D_ARRAY_TYPE_DEFINE` in the c file.

Reply:
I followed the structure pattern in TARRAY and TQUEUE, can I keep them consistent like this?

Reply:
ah, I see. Sure that's fine. It does seem like extra boilerplate needed to use this, but there may have been a good reason (maybe for tests/mocks), so let's keep the same design.